go/exocom/src/client_registry/client_registry.go

```diff
@@ -0,0 +1,77 @@
+package clientRegistry
+
+import "encoding/json"
+
+// Client is the combination of a client name, service type and internal namespace
+type Client struct {
+	ClientName        string
+	ServiceType       string
+	InternalNamespace string
+}
+
+// Route is an entry in the routes
+type Route struct {
```

Comment:
For the rewrite we just kept the same internals as in the js version. Let explore the subsystems when you get back as I'm unclear what exactly lies within each boundary